lib/scala/pkg/src/main/scala/org/enso/pkg/ConfigCodecs.scala

```diff
@@ -0,0 +1,69 @@
+package org.enso.pkg
+
+import io.circe._
+import io.circe.syntax._
+
+/** A collection of utility codecs used in the [[Config]]. */
+object ConfigCodecs {
+
+  /** The common decoding failure.
+    *
+    * @param entity the name of decoded entity
+    * @param history the list of JSON cursor operations
+    */
+  private def decodingFailure(
+    entity: String,
+    history: List[CursorOp]
+  ): DecodingFailure =
+    DecodingFailure(s"Failed to decode $entity", history)
+
+  /** Try to decode the entity `A` from a JSON object.
+    *
+    * @param entity the name of decoded entity
+    * @param nameKey the name key of the JSON object
+    * @param cursor the current focus in the JSON document
+    */
+  def getFromObject[A: Decoder](
+    entity: String,
+    nameKey: String,
+    cursor: HCursor
+  ): Decoder.Result[A] =
```

Comment:
I'm not sure if I understand what this function does. Is the `entity` name only used for pretty display of errors?